ignite/cmd/plugin.go

```diff
@@ -103,38 +103,45 @@ func parseGlobalPlugins() (cfg *pluginsconfig.Config, err error) {
 	return
 }
 
-func linkPlugins(rootCmd *cobra.Command, plugins []*plugin.Plugin) error {
+func linkPlugins(ctx context.Context, rootCmd *cobra.Command, plugins []*plugin.Plugin) error {
 	// Link plugins to related commands
 	var linkErrors []*plugin.Plugin
 	for _, p := range plugins {
 		if p.Error != nil {
 			linkErrors = append(linkErrors, p)
 			continue
 		}
-		manifest, err := p.Interface.Manifest()
+
+		manifest, err := p.Interface.Manifest(ctx)
 		if err != nil {
-			p.Error = fmt.Errorf("Manifest() error: %w", err)
+			p.Error = err
+			linkErrors = append(linkErrors, p)
```

Comment:
maybe we should use a `errgroup` here?

Reply:
I think it could be an improvement for a different PR if we think it would be a worthy change. It would require some refactor to make `linkPluginHooks` and `linkPluginCmds` stop when the group context is done for the async behaviour makes sense I believe. The current implementation prints all errors found while linking plugins and that won't happen if we introduce `errgroup` because the plugin linking gophers would stop when the first error is found.

Personally I like how they implemented the current behaviour, but it's true that the semantics here could be better.